typescript/web/src/components/labeling-tool/options-tool-bar/options-tool-bar.fixtures.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  GetLabelClassQuery,
+  GetLabelClassQueryVariables,
+} from "../../../graphql-types/GetLabelClassQuery";
+import {
+  GetLabelIdAndClassIdQuery,
+  GetLabelIdAndClassIdQueryVariables,
+} from "../../../graphql-types/GetLabelIdAndClassIdQuery";
+import {
+  GetLabelWithLabelClassQuery,
+  GetLabelWithLabelClassQueryVariables,
+} from "../../../graphql-types/GetLabelWithLabelClassQuery";
+import {
+  UpdateLabelClassActionMutation,
+  UpdateLabelClassActionMutationVariables,
+} from "../../../graphql-types/UpdateLabelClassActionMutation";
+import {
+  UpdateLabelClassOfLabelMutation,
+  UpdateLabelClassOfLabelMutationVariables,
+} from "../../../graphql-types/UpdateLabelClassOfLabelMutation";
+import {
+  ApolloMockResponse,
+  ApolloMockResponses,
+} from "../../../utils/tests/apollo-mock";
+import {
+  BASIC_LABEL_DATA,
+  DEEP_DATASET_WITH_CLASSES_DATA,
+} from "../../../utils/tests/data.fixtures";
+import {
+  GET_LABEL_CLASSES_OF_DATASET_QUERY,
+  GET_LABEL_QUERY as GET_LABEL_WITH_LABEL_CLASS_QUERY,
+  labelClassQuery,
+} from "../openlayers-map/queries";
+
+export const GET_LABEL_CLASSES_OF_DATASET_MOCK: ApolloMockResponse<
+  GetLabelClassesOfDatasetQuery,
+  GetLabelClassesOfDatasetQueryVariables
+> = {
+  request: {
+    query: GET_LABEL_CLASSES_OF_DATASET_QUERY,
+    variables: {
+      slug: DEEP_DATASET_WITH_CLASSES_DATA.slug,
+      workspaceSlug: DEEP_DATASET_WITH_CLASSES_DATA.workspace.slug,
+    },
+  },
+  result: {
+    data: {
+      dataset: {
+        id: DEEP_DATASET_WITH_CLASSES_DATA.id,
+        labelClasses: DEEP_DATASET_WITH_CLASSES_DATA.labelClasses.map(
+          (labelClass) => pick(labelClass, "id", "name", "color")
+        ),
+      },
+    },
+  },
+};
+
+export const GET_LABEL_WITH_LABEL_CLASS_MOCK: ApolloMockResponse<
+  GetLabelWithLabelClassQuery,
+  GetLabelWithLabelClassQueryVariables
+> = {
+  request: {
+    query: GET_LABEL_WITH_LABEL_CLASS_QUERY,
+    variables: { id: BASIC_LABEL_DATA.id },
+  },
+  nMatches: Number.POSITIVE_INFINITY,
+  result: {
+    data: {
+      label: {
+        ...pick(BASIC_LABEL_DATA, "id", "type"),
+        labelClass: pick(BASIC_LABEL_DATA.labelClass, "id", "name", "color"),
+      },
+    },
+  },
+};
+
+export const GET_LABEL_CLASS_MOCK: ApolloMockResponse<
+  GetLabelClassQuery,
+  GetLabelClassQueryVariables
+> = {
+  request: {
+    query: labelClassQuery,
+    variables: MATCH_ANY_PARAMETERS,
+  },
+  nMatches: Number.POSITIVE_INFINITY,
+  result: jest.fn((variables) => ({
+    data: {
+      labelClass:
+        variables.id === DEEP_DATASET_WITH_CLASSES_DATA.labelClasses[1].id
+          ? pick(
+              DEEP_DATASET_WITH_CLASSES_DATA.labelClasses[1],
+              "id",
+              "name",
+              "color"
+            )
+          : {
+              id: variables.id,
+              name: "uknown label class name",
+              color: "#123456",
+            },
+    },
+  })),
```

Comment:
This is starting to be complex and non-factorized (the `[1]` should be done only once). Can you move it into its own function(s)?